app/forms.py

```diff
@@ -0,0 +1,143 @@
+# import re
+from django import forms
+# from PIL import Image
+from django.contrib.auth.models import User
+from django.utils.translation import ugettext_lazy as _
+# from django.core.exceptions import ValidationError
+import time
+
+
+class RegistrationForm(forms.Form):
+
+    username = forms.RegexField(
+        regex=r'^\w+$',
+        widget=forms.TextInput(attrs=dict(required=True, max_length=30)),
+        label=_("Username"),
+        error_messages={'invalid': _("This value must contain only letters, \
+         numbers and underscores.")})
+    password1 = forms.CharField(
+        widget=forms.PasswordInput(
+            attrs=dict(
+                required=True,
+                max_length=4,
+                render_value=False,
+                type='password')),
+        label=_("PIN"))
```

Comment:
Why call this field `password1`? Was this copied & pasted from a form that asked for the password twice for validation perhaps?